app/classes/Transvision/Consistency.php

```diff
@@ -0,0 +1,121 @@
+<?php
+namespace Transvision;
+
+/**
+ * Consistency class
+ *
+ * Functions used to analyze consistency
+ *
+ * @package Transvision
+ */
+class Consistency
+{
+    /**
+     * Find duplicated strings (strings with the same text, ignoring case).
+     *
+     * @param array $strings_array Array of strings in the form
+     *                             string_id => string_value
+     *
+     * @return array Array of duplicated strings, same format as the
+     *               input array
+     */
+    public static function findDuplicates($strings_array)
+    {
+        $duplicates = [];
+        asort($strings_array);
+
+        $previous_key = '';
+        $previous_value = '';
+        foreach ($strings_array as $key => $value) {
+            if (strcasecmp($previous_value, $value) === 0) {
+                $duplicates[$previous_key] = $previous_value;
+                $duplicates[$key] = $strings_array[$key];
+            }
+            $previous_value = $value;
+            $previous_key = $key;
+        }
+
+        return $duplicates;
+    }
+
+     /**
+      * Filter out strings that should not be evaluted for consistency.
+      *
+      * @param  array $strings_array Array of strings in the form
+      *                              string_id => string_value
+      * @param  string Repository identifier
+      *
+      * @return array Array of filtered strings, with known false positives
+      *               removed
+      */
+     public static function filterStrings($strings_array, $repo)
+     {
+         if (in_array($repo, Project::getDesktopRepositories())) {
+             $repository_type = 'desktop';
+         } elseif (in_array($repo, Project::getGaiaRepositories())) {
+             $repository_type = 'gaia';
+         } else {
+             $repository_type = $repo;
+         }
+
+         $ignore_string = function ($key, $value) use ($repository_type) {
+             // Determine if a string should be excluded.
+
+             // Ignore strings of lenght 1 (e.g. accesskeys) and empty strings.
+             if (strlen($value) <= 1) {
+                 return true;
+             }
+
+             // Exclude CSS rules, "widht:" or "height:"
+             if (strpos($value, 'width:') !== false || strpos($value, 'height:') !== false) {
+                 return true;
+             }
+
+             // Exclude CSS widht values like '38em'
+             if (preg_match('/[\d|\.]+em/', $value)) {
+                 return true;
+             }
+
+             if ($repository_type == 'gaia') {
+                 // Ignore plural forms containing [] in the key
+                 if (strpos($key, '[') !== false && strpos($key, '[') !== false) {
```

Comment:
Did you mean to search for [ and ]? I think there is a typo since you are looking for [ twice. Also, would be another candidate for `` if (Strings::inString($value, ['[', ']']))```